crates/shared/src/price_estimation/oneinch.rs

```diff
@@ -0,0 +1,318 @@
+use super::gas;
+use crate::oneinch_api::{OneInchClient, RestResponse, SellOrderQuoteQuery};
+use crate::price_estimation::{Estimate, PriceEstimating, PriceEstimationError, Query};
+use anyhow::Result;
+use cached::{Cached, TimedSizedCache};
+use futures::future;
+use model::order::OrderKind;
+use primitive_types::U256;
+use std::sync::{Arc, Mutex};
+
+pub struct OneInchPriceEstimator {
+    api: Arc<dyn OneInchClient>,
+    disabled_protocols: Vec<String>,
+    allowed_protocols: Arc<Mutex<TimedSizedCache<Vec<String>, Vec<String>>>>,
```

Comment:
Since this logic is also used by the 1Inch solver, could we maybe move it to a shared component instead of re-implementing it here (I'm happy that `cached` is used instead of custom code!).

I'm fine with doing it in a separate PR to not have too much "scope creep" in this one.